fip-0052.md

```diff
@@ -0,0 +1,101 @@
+---
+fip: 52
+title: FIO Restake
+status: Draft
+type: Functionality
+author: Adam Androulidakis <adam@fio.net>
+created: 2025-05-18
+updated: 2025-07-14
+---
+
+# Abstract
+This FIP introduces a restaking mechanism for FIO rewards, enabling FIO handles to reinvest their staked rewards without the mandatory 7-day unlock period. In exchange, accounts pay a new premium to the network for the restake. A portion of this is allocated to the block producer (BP) reward pool, increasing compensation for BPs and addressing their growing need for revenue. 
+
+New actions
+|Contract|Action|Endpoint|Description|
+|---|---|---|---|
+|fio.token|restake|/retake_fio_tokens|Stakes specified FIO rewards.|
+
+## New getters
+|End point|Description|
+|---|---|
+|n/a|n/a|
+
+# Motivation
+
+The restaking mechanism allows FIO handles  to seamlessly compound their staking rewards by bypassing the 7-day unlock period, encouraging continuous participation and increasing the total FIO staked. The larger premium paid for restaking generates additional FIO for the network, which supports operational costs and protocol enhancements by directing a portion of these premiums to the BP reward pools. This mechanism enhances user convenience, boosts network revenue, and supports the BP reward pool.
+
+# Specification
+## Process
+  
+## New actions
+### Restake FIO
+Restakes FIO staking rewards without the 7 day unlock period
+#### Contract: fio.token
+#### New action: *restake*
+#### New end point: /restake_fio_tokens
+#### New fee: restake_fio_tokens, fee amount will be determined during development and updated here
+#### RAM increase: To be determined during implementation
+#### Request body
+|Parameter|Required|Format|Definition|
+|---|---|---|---|
+|fioaddress|Yes|FIO Address|FIO Address of the generating account. This will be used to verify domain ownership, and the NFT signature created for the address|
+|amount|Yes|Positive Int|Amount of rewards to restake|
+|max_fee|Yes|Positive Int|Maximum amount of SUFs the user is willing to pay for fee. Should be preceded by [/get_fee](https://dev.fio.net/reference/get_fee) for correct value.|
+|tpid|Yes|FIO Address|FIO Address of the entity which generates this transaction. TPID rewards will be paid to this address. Set to empty if not known. Auto-renew transactions will carry this TPID.|
+|actor|Yes|12 character string|Valid actor of signer|
+##### Example
+```
+{
+  "fioaddress": "1000000",
+  "amount": "1000000",
+  "max_fee": 30000000000,
+  "tpid": "rewards@wallet",
+  "actor": "aftyershcu22"
+}
+```
+#### Processing
+* FIO Address is validated
+* Amount parameter is validated
+* Request is validated per Exception handling
+* RAM of actor calling action is increased
+* restake_fio_tokens is charged to actor calling action
+* a votable portion of fees are issued from the accounts rewards to the block producer reward pools
```

Comment:
More detail definition is required here. What exactly is happening when user unstakes. The way that staking is designed is you earn SRPs when you stake, they appreciate in value while fees are collected, so when you unstake, the SRPs are converted to a higher amount of FIO. You would need to do some math here to figure out how that conversion happens and how that impacts your "principal". It's going to be tricky to figure this out. See [FIP-21](https://github.com/fioprotocol/fips/blob/master/fip-0021.md) for more details.